ipa-core/src/protocol/hybrid/oprf.rs

```diff
@@ -144,22 +175,148 @@ where
+                TestWorld::with_shards(TestWorldConfig {
+                    initial_gate: Some(Gate::default().narrow(&ProtocolStep::Hybrid)),
+                    ..Default::default()
+                });
+
+            let contexts = world.contexts();
+
+            let records = [
+                TestHybridRecord::TestImpression {
+                    match_key: 12345,
+                    breakdown_key: 2,
+                },
+                TestHybridRecord::TestImpression {
+                    match_key: 68362,
+                    breakdown_key: 1,
+                },
+                TestHybridRecord::TestConversion {
+                    match_key: 12345,
+                    value: 5,
+                },
+                TestHybridRecord::TestConversion {
+                    match_key: 68362,
+                    value: 2,
+                },
+                TestHybridRecord::TestImpression {
+                    match_key: 68362,
+                    breakdown_key: 1,
+                },
+                TestHybridRecord::TestConversion {
+                    match_key: 68362,
+                    value: 7,
+                },
+            ];
+            let indices_to_compare = [(0, 2), (1, 3), (1, 4), (1, 5)];
+
+            let shares: [Vec<HybridReport<BA8, BA3>>; 3] = records.iter().cloned().share();
+
+            let indistinguishable_reports: [Vec<IndistinguishableHybridReport<BA8, BA3>>; 3] =
+                shares
+                    .iter()
+                    .map(|s| s.iter().map(|r| r.clone().into()).collect::<Vec<_>>())
+                    .collect::<Vec<_>>()
+                    .try_into()
+                    .expect("Expected exactly 3 elements");
+
+            let chunked_reports: [Vec<Vec<IndistinguishableHybridReport<BA8, BA3>>>; 3] =
+                indistinguishable_reports
+                    .iter()
+                    .map(|vec| {
+                        let mid = vec.len() / SHARDS;
+                        vec.chunks(mid)
+                            .map(<[IndistinguishableHybridReport<BA8, BA3>]>::to_vec)
+                            .collect::<Vec<_>>()
+                    })
+                    .collect::<Vec<_>>()
+                    .try_into()
+                    .expect("Expected exactly 3 elements");
+
+            // #[allow(clippy::large_futures)]
+            // let results = chunked_reports
+            //     .into_iter()
+            //     .zip(contexts)
+            //     .map(|(reports_by_helper, helper_ctxs)| {
+            //         reports_by_helper
+            //             .into_iter()
+            //             .zip(helper_ctxs)
+            //             .map(|(reports, ctx)| async move {
+            //                 compute_prf_for_inputs(ctx, &reports).await.unwrap()
+            //             })
+            //             .collect::<Vec<_>>()
+            //     })
+            //     .collect::<Vec<_>>();
```

Comment:
I can't seem to get this to work. the loop approach below seems to work. it's a bit less ergonomic, but hopefully the loop acceptable for a test. happy to try any suggestions, otherwise I'll delete this.

Reply:
this test is really doing 70% of what `world.semi_honest()` already does. It is recommended to use it as it takes care of sharing and reconstructing the outputs. https://github.com/private-attribution/ipa/blob/277fbeb8332128aa52c58d16896fd377757c286a/ipa-core/src/protocol/ipa_prf/prf_eval.rs#L304 maybe a good place to start

Reply:
If I remember correctly, there we're two issues with that approach:
1. it wasn't clear if/how to work with sharded inputs
2. the prf isn't reconstructed

Reply:
it does support sharding https://github.com/private-attribution/ipa/blob/f587a37e6e9bd9b9eb8580cea9d6ea4914bda2f9/ipa-core/src/protocol/ipa_prf/shuffle/sharded.rs#L590

If you don't need to reconstruct, that makes it easier - just return it and you'll get 3 vectors of PRF values back